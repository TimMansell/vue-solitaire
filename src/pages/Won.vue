<template>
  <GameOverlay center-content show-logo data-test="won">
    <template #title> Congratulations, you win! </template>
    <template #msg>
      <GameSummary />
    </template>
    <template #buttons>
      <StartNewGameButton />
    </template>
  </GameOverlay>
</template>

<script>
import GameOverlay from '@/components/GameOverlay.vue';
import GameSummary from '@/components/GameSummary.vue';
import StartNewGameButton from '@/components/StartNewGameButton.vue';

export default {
  name: 'WonPage',
  components: {
    GameOverlay,
    StartNewGameButton,
    GameSummary,
  },
};
</script>
