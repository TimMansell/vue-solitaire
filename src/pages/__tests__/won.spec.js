import { describe, it, expect } from 'vitest';
import { shallowMount } from '@vue/test-utils';
import Won from '@/pages/Won.vue';

describe('Won.vue', () => {
  it('renders the component without crashing', () => {
    const wrapper = shallowMount(Won);

    expect(wrapper.isVisible()).toBe(true);
  });
});
