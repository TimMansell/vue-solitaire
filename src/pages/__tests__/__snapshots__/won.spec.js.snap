// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Won.vue matches snapshot 1`] = `
<gameoverlay-stub centercontent="true" showlogo="true" data-test="won"> Congratulations, you win! <gamesummary-stub></gamesummary-stub>
  <startnewgamebutton-stub></startnewgamebutton-stub>
</gameoverlay-stub>
`;
